### ISSUE TITLE
bug: config file generated by `--generate-default-config` contains removed and deprecated flags

### ISSUE THREAD

**Issue by DmitriyLewen** [aquasecurity/trivy#8043]
## Description
Trivy includes all flags into config file when using `--generate-default-config` command.
But Trivy returns error for this config file:
```
➜ trivy image   
2024-12-03T12:44:48-07:00       INFO    Loaded  file_path="trivy.yaml"
2024-12-03T12:44:48-07:00       ERROR   "--reset" was removed. Use "trivy clean --all" instead.
2024-12-03T12:44:48-07:00       FATAL   Fatal error     flag error: db flag error: unable to parse flag: removed flag ("--reset")
```

We need to exclude removed and obsolete flags from the generated configuration file.

**Comment by dupuyarc** [comment]
Although this bug was closed, I am seeing these errors in v0.58.2, and there were other problems with the generation that the original bug reporter didn't encounter, please see [my comment in discussion 8035](https://github.com/aquasecurity/trivy/discussions/8035#discussioncomment-11948140) for details, summary quoted below:

> The generated config file should work with the same version of trivy used to generate it.
>
> Also, the configuration generator should at least
>
> 1. Issue a warning that the default ignore file doesn't exist, or
> 2. Create the file if it doesn't exist.
> 3. Alternately trivy could consider non-existence of a configured ignore file a non-fatal error.
>
> My preference would be for (1) but any of these would be acceptable.
>
> Finally, the generated config file sets the option to generate a config file, which fails (since it now exists).

**Comment by DmitriyLewen** [comment]
Hi @dupuyarc
We didn't include this fix in [v0.58.2](https://github.com/aquasecurity/trivy/discussions/8243).

You can check the fix in the `main` branch or the canary build.

FYI, you can see the links between fixes and releases [here](https://github.com/aquasecurity/trivy/milestones)
